### ISSUE TITLE
Question regarding reward function for Staircase

### ISSUE THREAD

**Issue by vzhong** [facebookresearch/nle#90]
Hey guys :)

Congrats on the NeurIPS acceptance!

I have a quick question regarding the reward graph from the paper (Fig 4). In this figure it seems like the running average for the reward ends up being something like 90-ish per episode. However the reward function for `NetHackStaircase` defines the reward as 1 if the agent descends and 0 otherwise. It seems then like the maximum reward (accounting for penalties) would be 1 per episode. How would get get 90 per episode from this reward function? Is this computed some other way?

As a related note, I've been trying to reproduce the results from the paper using the default hyperparameters and the baseline agent using `val-dwa-law-fem`. So far the `Scout` performance seems to be improving (e.g. 280/episode after 3e7 steps), however the `Staircase` reward seems to be plateauing at 0 after 3e7 steps. I know that the plots were produced using polybeast but just curious if y'all had some insights as to whether something is wrong here.

**Comment by heiner** [comment]
Hey Victor,

Thanks for your question. We'll answer shortly but I think the short answer is that we set a reward of 100 or so for finding the staircase in our paper experiments (this is something we definitely should mention so thanks for pointing this out just in time!). I'll assign to @rockt for further comment.

As for the agent in NLE, it's really only meant as a code example of how to interact with NLE. It does not and will never reproduce our baselines. We are working on releasing our full experimental code but that takes a while as it's ... well, rather experimental.

**Comment by aleSuglia** [comment]
Thanks @vzhong for opening this issue. I was actually going to ask something very similar. I would like to reproduce your full set of results from the NeurIPS paper and I was wondering whether you could report the commands used to train the models. It would be really useful to fully reproduce your setup. In the meantime, is there any way to reproduce your results by hacking the code and modify it myself? What is the delta between the model released and the one in the paper? 

Thanks a lot,
Alessandro

**Comment by vzhong** [comment]
@aleSuglia I believe one of the key difference is the use of polybeast (paper) instead of monobeast (this repo). Due to the parallel nature of the former, I believe the learning characteristics of the latter is different. FWIW, I have ran each task individually with the default hyperparameters. They all seem to improve (I didn't run them for 1 billion steps as in the paper), however the rate of convergence seems slower than the paper (perhaps due to different code base). The two envs that I find problematic are Staircase and Pets, neither of which converge to anything beyond zero even after I increased the reward from 1 to 100.
